assert/assertions.go

```diff
@@ -2010,15 +2010,30 @@ func Eventually(t TestingT, condition func() bool, waitFor time.Duration, tick t
 		h.Helper()
 	}
 
-	ch := make(chan bool, 1)
-	checkCond := func() { ch <- condition() }
+	const failed = 0
+	const stop = 1
+	const noStop = 2
+
+	resultCh := make(chan int, 1)
+	checkCond := func() {
+		result := failed
+		defer func() {
+			resultCh <- result
```

Comment:
For cleaner test results, we could also catch panics here and make the two failure modes `failed` and `panic` explicit with separate error messages. See my other variant [here](https://github.com/ubunatic/go-contrib-testify/blob/b9276033b772155ac653fa542e31ea89ff72d52c/assert/assertions.go#L2032).

Reply:
We should probably not change this to capture panics in the condition function, getting very close to a breaking change in the behavior.

Reply:
I agree.